### ISSUE TITLE
select part of screen and insert it as screenshot in document

### ISSUE THREAD

**Issue by Quoteme** [xournalpp/xournalpp#2747]
**Is your feature request related to a problem? Please describe.**
I wanted to copy a part of my screen into the document. For example when doing my homework, I could simply screenshot my tasks without needing to save each screenshot as a seperate picture.

**Describe the solution you'd like**
Adding some button to the UI that would execute `maim -s | xclip -selection clipboard -t image/png` and then paste it would be nice!

**Describe alternatives you've considered**
I have written a plug-in that works just like that, but it lacks a dedicated ui-button, so one first needs to open: menu > plugin > "Screenshot to clipboard". Maybe someone knows how to add this as a ui-button, or maybe this can be baked into standard xournalpp?

**Additional context**
Here is the code for anyone who wants to simply copy part of their screen with no hastle:

1. install xclip and maim
2. `sudo mkdir /usr/share/xournalpp/plugins/Screenshoty`
3. `cd /usr/share/xournalpp/plugins/Screenshoty`
4. `sudo touch plugin.ini main.lua`
5. paste the following code into these files:

plugin.ini:
```
[about]
## Author / Copyright notice
author=Luca Leon Happel

description=Select a portion of the screen and copy it to your clipboard (using maim)

## If the plugin is packed with Xournal++, use
## <xournalpp> then it gets the same version number
version=<xournalpp>

[default]
enabled=false

[plugin]
mainfile=main.lua
```

main.lua:
```
function initUi()
  app.registerUi({["menu"] = "Screenshot to clipboard", ["callback"] = "shot"});
end

function shot()
  os.execute("maim -s | xclip -selection clipboard -t image/png")
end
```

**Comment by personalizedrefrigerator** [comment]
There's [a plugin bundled with Xournal++ that does this](https://github.com/xournalpp/xournalpp/blob/master/plugins/QuickScreenshot/main.lua), but I don't think supports `xclip`! Would you be willing to open a PR that updates it?

**Comment by nicolae-stroncea** [comment]
@Quoteme you can use the accelerator argument to add a keyboard shortcut to trigger it:
```lua
function initUi()
    app.registerUi({["menu"] = "Screenshot to clipboard", ["callback"] = "shot", ["accelerator"] = "<Shift><Alt>e"});
end
```
This seems to be different from the currently bundled plugin. From my understanding, the bundled plugin is meant to take screenshots and save them to a file(so it can save parts of a xournal document as png). This copies a screenshot to clipboard instead so it can be pasted into a xournal document. 

Would be great to see this integrated as another plugin in the main repo.

**Comment by Quoteme** [comment]
Yeah, I can look into it once I am home!

**Comment by niyumard** [comment]
> > There's [a plugin bundled with Xournal++ that does this](https://github.com/xournalpp/xournalpp/blob/master/plugins/QuickScreenshot/main.lua), but I don't think supports `xclip`! Would you be willing to open a PR that updates it?
> 
> Yeah, I can look into it once I am home!

I'm facing the same need.
Adding this option to the plugin will be appreciated a lot.

**Comment by Quoteme** [comment]
> > > There's [a plugin bundled with Xournal++ that does this](https://github.com/xournalpp/xournalpp/blob/master/plugins/QuickScreenshot/main.lua), but I don't think supports `xclip`! Would you be willing to open a PR that updates it?
> > 
> > 
> > Yeah, I can look into it once I am home!
> 
> I'm facing the same need. Adding this option to the plugin will be appreciated a lot.

I just created a pull request (#3718) to use maim and xclip if these programs are installed. Make sure to install them.

**Comment by camoz** [comment]
I would also find it useful to have it paste the screenshot instantly, for when you have a convertible laptop in tablet mode (and don't want to click through Edit->Paste).

I tried adding:

```
app.uiAction({["action"] = "ACTION_PASTE"})
```

but for some reason that it does not work on wayland... but under X11 it works.

**Comment by camoz** [comment]
The requested toolbar button can be done now with https://github.com/xournalpp/xournalpp/pull/4671.

**Comment by amitnovick** [comment]
> The requested toolbar button can be done now with #4671.

@rolandlo 

Is this feature available in a release?
I'm currently running Xournal++ version 1.2.1 , however I don't have an option to add a toolbar icon for QuickScreenShot plugin through the Toolbars > Customize UI.

Perhaps QuickScreenShot plugin needs to register a toolbar icon for this?
https://github.com/xournalpp/xournalpp/tree/master/plugins/QuickScreenshot

**Comment by rolandlo** [comment]
Yes, this feature is available in the 1.2.1 release (and also in the 1.2.0 release).

> Perhaps QuickScreenShot plugin needs to register a toolbar icon for this?
> https://github.com/xournalpp/xournalpp/tree/master/plugins/QuickScreenshot

Yes, the plugin must register a toolbar icon with the callback, which the QuickScreenshot plugin does not. Are you suggesting to add one for the shipped plugin? What icon would you choose?

**Comment by amitnovick** [comment]
> Yes, this feature is available in the 1.2.1 release (and also in the 1.2.0 release).
> 
> > Perhaps QuickScreenShot plugin needs to register a toolbar icon for this?
> > https://github.com/xournalpp/xournalpp/tree/master/plugins/QuickScreenshot
> 
> Yes, the plugin must register a toolbar icon with the callback, which the QuickScreenshot plugin does not. Are you suggesting to add one for the shipped plugin? What icon would you choose?

Yes, I do suggest adding a toolbar icon for the QuickScreenshot plugin.

The rationale is as following:
- Obviously, having a quick-access button for it (on the toolbar) would be a QoL improvement over accessing it through the Plugins menu
- The toolbar is also where I intuitively think it should be (it's a tool)
- Moreover, the Plugins menu is not accessible in Full-screen mode

From brief glance at the icons folder (I presume), I conclude that SVG is the chosen file format for icons.
Does the chosen icon need to be provided under a specific license?
I am not an expert in SVG icon creation (even though I could dabble in it with Inkscape).
Something similar to
![image](https://github.com/xournalpp/xournalpp/assets/43674238/e097ddc2-35dd-437e-bbc0-b762b89a7c95)

seems appropriate (not my work, that's by https://thenounproject.com/icon/screenshot-3244586/)

What is the course of action in this scenario? (perhaps you've added icons to the project before?)

**Comment by rolandlo** [comment]
As things currently work we could simply put that icon into the `ui/iconsColor-light/hicolor/scalable/actions/` folder (and an inverted one for dark theme). It would get installed at the same place like the other icons (in the [shared resources folder](https://xournalpp.github.io/guide/file-locations/) and could then be accessed like any of the other icons and system icons.


However we plan to move all plugins into a separate repository. 
When the plugins get installed separately, how will it install the icons to the shared resources folder of `xournalpp` or to another place where the icons can be found?

Of course we could ask users to install the icon(s) into `~/.local/share/icons/hicolor/scalable/actions/` on Linux and respective places on Windows and MacOS. But that wouldn't be automatic. So I don't see a good solution right now for shipping plugin icons.
